builder/vsphere/iso/step_create.go

```diff
@@ -18,6 +18,17 @@ import (
 	"github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver"
 )
 
+type vAppConfig struct {
```

Comment:
Is there a reason why `vAppConfig` is introduced here? If we need the properties only, we could introduce a `vapp_properties` attribute to the main config directly rather than as a separate attribute.
Also I may be mistaken by I believe `packer-sdc` only considers public types for generation, I wonder how the flat configs were generated in the first place? But this would explain the `make generate` error we see in the CI as it removes the structure from the generated code.

Reply:
This is how it's implemented `vsphere-clone` builder.

I just pushed a change that should hopefully resolve the CI. 🤞🏻

Reply:
https://github.com/hashicorp/packer-plugin-vsphere/pull/299#issuecomment-1802409752